src/containers/analyze/upload/components/upload.tsx

```
@@ -7,29 +7,39 @@ import {
 import type { UploadProps } from "antd";
 import { InboxOutlined } from "@ant-design/icons";
 import { message } from "antd";
-
 import { useHistory } from "react-router-dom";
 import { API_ENDPOINT } from "../../../../utils/constants.ts";
 
 const Uploader = () => {
   const { push } = useHistory();
+  const MAX_FILE_SIZE = 1 * 1024 * 1024 * 1024; // 1GB in bytes
 
   const props: UploadProps = {
     name: "video",
     multiple: false,
     method: "post",
     action: `${API_ENDPOINT}/analyze_video`,
     listType: "picture",
-    accept: ".mp4,.mov,.avi,.wmc,.avchd,.flv,.webm,.mkv,.mpeg,.vob,.wmv", // Adding the accept attribute here
+    accept: ".mp4,.mov,.avi,.wmc,.avchd,.flv,.webm,.mkv,.mpeg,.vob,.wmv",
+    beforeUpload: (file) => {
+      if (file.size > MAX_FILE_SIZE) {
+        message.error(
+          `This file is too large. File size must be less than 1GB.`
+        );
+        return Upload.LIST_IGNORE; // Prevent the file from being uploaded
+      }
+      setTimeout(() => {
+        // Code to hide the error message
```

is this a placeholder?

---

Yes, it is a placeholder for a setTimeout function. It includes a comment indicating where the actual code to hide an error message should be placed but for now not used for that purpose. we use the setTimeout function to set and execute after 15,000 milliseconds or 15 seconds